packages/labs/task/src/test/task_test.ts

```diff
@@ -137,27 +103,129 @@ suite('Task', () => {
     }
   });
 
-  test('initial status', async () => {
-    el = new A();
-    assert.equal(el.task.value, undefined);
+  test('task without args do not run', async () => {
+    const el = await renderElement(getTestElement());
+    assert.equal(el.task.status, TaskStatus.INITIAL);
+    assert.equal(el.taskValue, undefined);
+    el.requestUpdate();
+    await tasksUpdateComplete();
     assert.equal(el.task.status, TaskStatus.INITIAL);
-    assert.equal(el.taskController.value, undefined);
-    assert.equal(el.taskController.status, TaskStatus.INITIAL);
+    assert.equal(el.taskValue, undefined);
   });
 
-  test('pending and complete status', async () => {
+  test('tasks with args run initially', async () => {
```

Comment:
I think you need a test for `{args: () => []}`. Seems like that might not run with the current implementation of `_argsDirty`.

Reply:
Added.